PWGHF/TableProducer/candidateCreatorToXiPi.cxx

```diff
@@ -468,238 +151,191 @@ struct HfCandidateCreatorToXiPi {
-
-        // set hfFlag
-        int hfFlag = 1 << aod::hf_cand_toxipi::DecayType::DecayToXiPi;
-
-        // fill test histograms
-        hInvMassCharmBaryon->Fill(mCharmBaryon);
-        hCandidateCounter->Fill(3);
-
-        // fill the table
-        rowCandidate(collision.globalIndex(),
-                     pvCoord[0], pvCoord[1], pvCoord[2],
-                     vertexCharmBaryonFromFitter[0], vertexCharmBaryonFromFitter[1], vertexCharmBaryonFromFitter[2],
-                     vertexCasc[0], vertexCasc[1], vertexCasc[2],
-                     vertexV0[0], vertexV0[1], vertexV0[2],
-                     trackXiDauCharged.sign(),
-                     chi2PCACharmBaryon, covVtxCharmBaryon[0], covVtxCharmBaryon[1], covVtxCharmBaryon[2], covVtxCharmBaryon[3], covVtxCharmBaryon[4], covVtxCharmBaryon[5],
-                     pVecCharmBaryon[0], pVecCharmBaryon[1], pVecCharmBaryon[2],
-                     pVecCasc[0], pVecCasc[1], pVecCasc[2],
-                     pVecPionFromCharmBaryon[0], pVecPionFromCharmBaryon[1], pVecPionFromCharmBaryon[2],
-                     pVecV0[0], pVecV0[1], pVecV0[2],
-                     pVecPionFromCasc[0], pVecPionFromCasc[1], pVecPionFromCasc[2],
-                     pVecV0Dau0[0], pVecV0Dau0[1], pVecV0Dau0[2],
-                     pVecV0Dau1[0], pVecV0Dau1[1], pVecV0Dau1[2],
-                     impactParameterCasc.getY(), impactParPiFromCharmBaryonXY,
-                     impactParameterCasc.getZ(), impactParPiFromCharmBaryonZ,
-                     std::sqrt(impactParameterCasc.getSigmaY2()), std::sqrt(impactParameterPiFromCharmBaryon.getSigmaY2()),
-                     v0Element.globalIndex(), v0Element.posTrackId(), v0Element.negTrackId(),
-                     casc.globalIndex(), trackPion.globalIndex(), trackXiDauCharged.globalIndex(),
-                     mLambda, mCasc, mCharmBaryon,
-                     cpaV0, cpaCharmBaryon, cpaCasc, cpaxyV0, cpaxyCharmBaryon, cpaxyCasc,
-                     ctOmegac, ctCascade, ctV0, ctXic,
-                     pseudorapV0Dau0, pseudorapV0Dau1, pseudorapPiFromCas, pseudorapPiFromCharmBaryon,
-                     pseudorapCharmBaryon, pseudorapCascade, pseudorapV0,
-                     dcaxyV0Dau0, dcaxyV0Dau1, dcaxyPiFromCasc,
-                     dcazV0Dau0, dcazV0Dau1, dcazPiFromCasc,
-                     dcaCascDau, dcaV0Dau, dcaCharmBaryonDau,
-                     decLenCharmBaryon, decLenCascade, decLenV0, errorDecayLengthCharmBaryon, errorDecayLengthXYCharmBaryon,
-                     hfFlag);
-
-      } // loop over LF Cascade-bachelor candidates
-    }   // loop over collisions
-  }     // end of process
-  PROCESS_SWITCH(HfCandidateCreatorToXiPi, processDerivedData, "Process derived data", false);
-
-}; // end of struct
+      auto trackPion = cand.prong0_as<TracksWCovDca>();
+      auto cascElement = cand.cascade_as<aod::CascadesLinked>();
+      auto casc = cascElement.cascData_as<MyCascTable>();
```

Comment:
Since you never use `cascElement` anymore and it can confuse a non-expert reader (like me), what about changing into ``` auto casc = cand.cascade_as<aod::CascadesLinked>().cascData_as<MyCascTable>(); ``` ?

Reply:
Hi @mfaggin, I think it would be nice to clearly separate v0/cascades from aod tables and from PWG tables. I agree that the name is not really explicative, I can change it to something more suitable, maybe v0/cascAodElement